rudder-lang/src/technique.rs

```diff
@@ -0,0 +1,338 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: 2019-2020 Normation SAS
+
+use crate::{
+    ast::AST,
+    error::*,
+    io::IOContext,
+    rudderlang_lib::{RudderlangLib, LibMethod},
+    cfstrings,
+};
+use serde::{Deserialize, Serialize};
+use std::convert::From;
+use colored::Colorize;
+use std::{fs, str};
+use lazy_static::lazy_static;
+use regex::{Regex, Captures};
+use typed_arena::Arena;
+
+// Techniques are limited subsets of CFEngine in JSON representation
+// that only carry method calls and Rudder metadata
+/// generates a technique to either rudderlang or json format, using our own library
+pub fn generate(context: &IOContext) -> Result<()> {
+    let sources: Arena<String> = Arena::new();
+    let lib = RudderlangLib::new(&context.stdlib, &sources)?;
+    let technique = Technique::new(&context)?;
+    let technique_as_rudderlang = technique.to_rudderlang(&lib)?;
+    let output_path = context.dest.to_string_lossy();
+    
+    // will disapear soon, return string directly
+    fs::write(&context.dest, technique_as_rudderlang)
+        .map_err(|e| err_wrapper(&output_path, e))?;
+    Ok(())
+}
+
+/// Every Technique substructure has only 1 purpose: represent a Technique as json or rudderlang string
+#[derive(Serialize, Deserialize)]
+pub struct Technique {
+    r#type: String,
+    version: usize,
```

Comment:
should probably be a string for us (usize is not good as it could be `1.1`, but we probably don't need a better representation for what we use it for)